src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -195,7 +196,7 @@ private static TemplateMatcher GetInstanceRaiseEventRoute()
                         status.RuntimeStatus == OrchestrationRuntimeStatus.Failed ||
                         status.RuntimeStatus == OrchestrationRuntimeStatus.Terminated)
                     {
-                        return await this.HandleGetStatusRequestAsync(request, instanceId);
+                        return await this.HandleGetStatusRequestAsync(request, instanceId, returnInternalServerErrorOnFailure);
                     }
                 }

```

Comment:
It's not featured in this code segment, but I believe we need to also provide `returnInternalServerErrorOnFailure` in our call to CreateCheckStatusResponseMessage below.

Reply:
Why/Where do I need to do that? Are you talking about the call 12 lines down? I actually don't need to add it there because the HttpManagementPayload with the URIs was created from GetClinetResponseLinks on line 181.

Reply:
Ah, I just misunderstood that code segment. We are all good here.